src/KeyVault/KeyVault/Commands/BackupAzureKeyVaultKey.cs

```diff
@@ -43,69 +44,103 @@ public class BackupAzureKeyVaultKey : KeyVaultCmdletBase
+        [Parameter(Mandatory = true,
+            Position = 1,
+            ParameterSetName = HsmByKeyNameParameterSet)]
+        [ValidateNotNullOrEmpty]
         [Alias(Constants.KeyName)]
         public string Name { get; set; }
 
         /// <summary>
         /// KeyBundle object to be backed up.
         /// </summary>
         /// <remarks>
-        /// Note that the backup applies to the entire family of a key (current and all its versions); 
+        /// Note that the backup applies to the entire family of a key (current and all its versions);
         /// since a key bundle represents a single version, the intent of this parameter is to allow pipelining.
         /// The backup cmdlet will use the Name and VaultName properties of the KeyBundle parameter.
         /// </remarks>
         [Parameter(Mandatory = true,
-                    Position = 0,
-                    ValueFromPipeline = true,
-                    ParameterSetName = ByKeyObjectParameterSet,
-                    HelpMessage = "Key bundle to back up, pipelined in from the output of a retrieval call.")]
+            Position = 0,
+            ValueFromPipeline = true,
+            ParameterSetName = ByKeyObjectParameterSet,
+            HelpMessage = "Key bundle to back up, pipelined in from the output of a retrieval call.")]
         [ValidateNotNullOrEmpty]
         [Alias("Key")]
         public PSKeyVaultKeyIdentityItem InputObject { get; set; }
 
         /// <summary>
         /// The output file in which the backup blob is to be stored
         /// </summary>
-        [Parameter( Mandatory = false,
-                    Position = 2,
-                    HelpMessage = "Output file. The output file to store the backed up key blob in. If not present, a default filename is chosen.")]
+        [Parameter(Mandatory = false,
+            Position = 2,
+            HelpMessage = "Output file. The output file to store the backed up key blob in. If not present, a default filename is chosen.")]
         [ValidateNotNullOrEmpty]
         public string OutputFile { get; set; }
 
         /// <summary>
         /// Instructs the cmdlet to overwrite the destination file, if it exists.
         /// </summary>
-        [Parameter( Mandatory = false,
-                    HelpMessage = "Overwrite the given file if it exists" )]
+        [Parameter(Mandatory = false,
+            HelpMessage = "Overwrite the given file if it exists")]
         public SwitchParameter Force { get; set; }
 
         #endregion Input Parameter Definition
 
         public override void ExecuteCmdlet()
+        {
+            NormalizeParameterSets();
+
+            if (string.IsNullOrEmpty(HsmName))
+            {
+                BackupKeyVaultKey();
+            }
+            else
+            {
+                BackupHsmKey();
+            }
+        }
+
+        private void NormalizeParameterSets()
```

Comment:
You can put NormalizeParameterSets into base class

Reply:
I wish I could, but the logic of this method is slightly different in every cmdlet.
For example, some cmdlets have `-ResourceId` and some don't, some even have `-HsmResourceId` as well.
I'm afarid we won't get much benifit to write it in base class.